test/burn/org/apache/cassandra/net/ConnectionBurnTest.java

```diff
@@ -467,26 +491,42 @@ public void run() throws ExecutionException, InterruptedException, NoSuchFieldEx
                     }
                 });
 
-                while (deadline > nanoTime() && failed.getCount() > 0)
+                int added = 0;
+                while (deadline > nanoTime() && added == 0)
                 {
                     reporters.update();
                     reporters.print();
-                    Uninterruptibles.awaitUninterruptibly(failed, 30L, TimeUnit.SECONDS);
+                    added = Queues.drainUninterruptibly(unexpectedExceptions, exceptions, 10, 30L, TimeUnit.SECONDS);
```

Comment:
why 10 items, rather than 1?

Reply:
Fixed, and move this value to a constant value for visibility.